### PR TITLE
Bucky/changelog

### PR DESCRIPTION
**Pull request by ebuchman**
Updated the changelog to add PR refs and some things I found to be missing.

Need to add a couple more things but have to run for now.

Also changed the "BREAKING" section to have subsections so we can be more clear about what broke and who should care. Here's a direct link: https://github.com/tendermint/tendermint/blob/bucky/changelog/CHANGELOG_PENDING.md

Thoughts?

### CODE REVIEW

File: CHANGELOG_PENDING.md [hunk 1]
```diff
@@ -1,45 +1,67 @@
...
+- [mempool] \#2234 Now stores txs by hash inside of the cache, to mitigate memory leakage
 
 BUG FIXES:
-- [mempool] No longer possible to fill up linked list without getting caching
-  benefits [#2180](https://github.com/tendermint/tendermint/issues/2180)
-- [state] kv store index tx.height to support search
-- [rpc] /dial_peers does not try to dial existing peers
+- [config] \#2284 Replace `db_path` with `db_dir` from automatically generated configuration files.
+- [mempool] \#2188 Fix OOM issue from cache map and list getting out of sync
+- [state] #2051 KV store index supports searching by `tx.height`
```
Comment (melekes) [inline CHANGELOG_PENDING.md]: `\` is missing before `#2051`

File: CHANGELOG_PENDING.md [hunk 2]
```diff
@@ -1,45 +1,67 @@
...
+    - [types] Updates to the Header:
+        - \#1815 NextValidatorsHash
+        - \#2106 ProposerAddress
+    - [blockchain] \#2213 Fix Amino routes for blockchain reactor messages
+    - [state] \#1815 Validator set changes are delayed by one block (!)
+        - Add NextValidatorSet to State, changes on-disk representation of state
+    - [consensus] \#2203 Implement BFT time
+        - Timestamp in block must equal median of timestamps in block's LastCommit
+    - [p2p] \#2263 Update secret connection to use a little endian encoded nonce
+    - [crypto] \#2205 Rename AminoRoute variables to no longer be prefixed by signature type.
```
Comment (ValarDragon) [inline CHANGELOG_PENDING.md]: This is a Go API thing, the actual routes were unchanged

File: CHANGELOG_PENDING.md [hunk 3]
```diff
@@ -1,45 +1,67 @@
...
+    - [crypto] \#2239 Secp256k1 signature changes (See ADR-014):
```
Comment (ValarDragon) [inline CHANGELOG_PENDING.md]: Can we link to ADR-014 as well?

File: CHANGELOG_PENDING.md [hunk 4]
```diff
@@ -1,45 +1,67 @@
...
-- [types] allow genesis file to have 0 validators ([#2015](https://github.com/tendermint/tendermint/issues/2015))
-- [libs] allow passing options through when creating instances of leveldb dbs ([#2292](https://github.com/tendermint/tendermint/issues/2292))
+- [types] Allow genesis file to have 0 validators ([#2015](https://github.com/tendermint/tendermint/issues/2015))
+    - Initial validator set can be determined by the app in ResponseInitChain
+- [libs/db] \#2293 Allow passing options through when creating instances of leveldb dbs
 
 IMPROVEMENTS:
 - [docs] Lint documentation with `write-good` and `stop-words`.
 - [scripts] Added json2wal tool, which is supposed to help our users restore
   corrupted WAL files and compose test WAL files (@bradyjoestar)
```
Comment (ValarDragon) [inline CHANGELOG_PENDING.md]: Is suffixing all community contributions with who wrote something we're doing? I don't think I've seen this in changelogs before, but I'm also not opposed.

File: CHANGELOG_PENDING.md [hunk 5]
```diff
@@ -1,45 +1,67 @@
...
+    - [crypto] \#2205 Rename AminoRoute variables to no longer be prefixed by signature type.
+    - [crypto] \#2239 Secp256k1 signature changes (See ADR-014):
+        - format changed from DER to `r || s`, both little endian encoded as 32 bytes.
+        - malleability removed by requiring `s` to be in canonical form.
+
 
 FEATURES:
-- [types] allow genesis file to have 0 validators ([#2015](https://github.com/tendermint/tendermint/issues/2015))
-- [libs] allow passing options through when creating instances of leveldb dbs ([#2292](https://github.com/tendermint/tendermint/issues/2292))
+- [types] Allow genesis file to have 0 validators ([#2015](https://github.com/tendermint/tendermint/issues/2015))
```
Comment (ValarDragon) [inline CHANGELOG_PENDING.md]: Should we make #2015 follow the same format as everything else? (i.e. be at the front)

**Comment by ValarDragon** [comment]
In the spirit of p2p having its own version (per your ADR), I think it should have its own changelog section instead of being a part of "Protocol"

File: CHANGELOG_PENDING.md (hunk 4 above)
Comment (ebuchman) [reply CHANGELOG_PENDING.md]: Not sure how best to do this but I really would love to give shout outs to open source contributors in the changelog

**Comment by ebuchman** [comment]
Thanks for the review folks - I addressed everything and added a couple more points we forgot (including max bytes stuff!)

**Comment by ValarDragon** [comment]
We probably should make a script that takes all regex matches for `\#<3 or 4 numbers>` and builds the correct PR / issue link

**Pull request merged**